numba/core/ssa.py

```diff
@@ -241,12 +255,26 @@ def on_other(self, states, stmt):
 
 
 class _GatherDefsHandler(_BaseHandler):
-    """Find all defs
+    """Find all defs and uses of variable in each block
 
-    ``states`` is a Mapping[str, List[ir.Assign]]
+    ``states["label"]`` is a int; label of the current block
+    ``states["defs"]`` is a Mapping[str, List[Tuple[ir.Assign, int]]]
```

Comment:
Added more comment at 55028233b26eb9adf53015d9722fbf240ed423ba.

> does states need to be a dict, would a namedtuple work, looks like it's just a container of a consistent 3 things?

it can be a namedtuple with some refactoring. For now, everything `_BaseHandler` subclass has different requirements for what is in `states`.

Reply:
ah ok, let's leave this as-is then, thanks for looking.